### PR TITLE
HPCC-28638 New base build image creation

### PR DESCRIPTION
**Pull request by GordonSmith**
ubuntu-22.04
ubuntu-20.04
ubuntu-18.04
centos-8
centos-7

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith please can you look through this change
@GordonSmith the jira could do with some details describing what the change is doing (and why).

File: dockerfiles/vcpkg/core.dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+WORKDIR /hpcc-dev
+
+COPY ./*.deb .
+
+RUN apt-get install --no-install-recommends -f -y ./*.deb
+
+EXPOSE 8010
+
+ENTRYPOINT ["touch /var/log/HPCCSystems/myesp/esp.log && \
```
Comment (jakesmith) [inline dockerfiles/vcpkg/core.dockerfile:15]: question: why is the touch necessary?

**Comment by jakesmith** [comment]
>  @GordonSmith the jira could do with some details describing what the change is doing (and why).

Yes, it's quite a big change/commit, it would help to know more detail and overview on what it is acheiving.

**Review by jakesmith — COMMENTED** [review]
@GordonSmith - a few minor initialy comments/questions.

File: CMakeLists.txt [hunk 2]
```diff
@@ -67,14 +67,17 @@
 #
 #########################################################
 
-cmake_minimum_required(VERSION 3.24.0)
+cmake_minimum_required(VERSION 3.22.1)
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply CMakeLists.txt:70]: The 3.24.0 version matched what vcpkg was using.
The 3.22.1 version matches what Ubuntu 22.04 will install from apt repo
So I tested with 3.22.1 and it appears to work fine (so will save folks some pain).

File: dockerfiles/vcpkg/centos-7.dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+FROM hpccbuilds/vcpkg-centos-7:$VCPKG_REF
+
+RUN yum install -y \
+    java-11-openjdk-devel \
+    python3-devel 
+
+WORKDIR /hpcc-dev
+
+ENV CMAKE_OPTIONS="\
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo \
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply dockerfiles/vcpkg/centos-7.dockerfile:11]: Th plan is to use RelWithDebInfo and produce two packages, one with stripped symbols and the other with symbols (hopefully that means stack dumps can be loaded into the latter for post mortem debugging)?

File: dockerfiles/vcpkg/core.dockerfile (hunk 1 above)
Comment (GordonSmith) [reply dockerfiles/vcpkg/core.dockerfile:15]: Whoops that file shouldn't be in this PR...

Comment (GordonSmith) [reply dockerfiles/vcpkg/core.dockerfile]: @jakesmith - I removed this file

**Comment by GordonSmith** [comment]
@ghalliday @jakesmith ~~ChatGPT~~ I added a description to the ticket.

**Review by jakesmith — COMMENTED** [review]
@GordonSmith - is this ready for review? I saw/see all the WIP commits, was waiting for them to die down/be squashed before looking further.

**Comment by GordonSmith** [comment]
Closing this PR in favour of a few smaller ones.  Also I have extracted the tag/asset creation into its own simpler GH action - thx for comments.

**Pull request closed without merging**